### PR TITLE
EVG-2159: add debug option to makefile

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -34,7 +34,7 @@ distTestRaceContents := $(foreach pkg,$(packages),$(buildDir)/race.$(pkg))
 srcFiles := makefile $(shell find . -name "*.go" -not -path "./$(buildDir)/*" -not -name "*_test.go" -not -path "./scripts/*" -not -path "*\#*")
 testSrcFiles := makefile $(shell find . -name "*.go" -not -path "./$(buildDir)/*" -not -path "*\#*")
 currentHash := $(shell git rev-parse HEAD)
-ldFlags := "-w -s -X=github.com/evergreen-ci/evergreen.BuildRevision=$(currentHash)"
+ldFlags := "$(if $(ENABLE_DEBUG),-w -s ,)-X=github.com/evergreen-ci/evergreen.BuildRevision=$(currentHash)"
```
Comment (bsamek) [inline makefile]: Should this be `$(if $(ENABLE_DEBUG),,-w -s)`, since if ENABLE_DEBUG is set, we *don't* want to disable DWARF/symbol table?

Comment (tychoish) [reply makefile]: This is what we do for the race option:
``$(if $(RACE_ENABLED),-race ,)``

here's my output: 

```[derrida ~/xgen/evergreen](EVG-2159-add-debug-option-to-makefile) % make -n cli  
mkdir -p bin
GOOS="" GOARCH="" go build -o bin/build-cross-compile scripts/build-cross-compile.go
echo go build -o bin/build-cross-compile scripts/build-cross-compile.go
./bin/build-cross-compile -buildName=linux_amd64 -ldflags="-X=github.com/evergreen-ci/evergreen.BuildRevision=72038235f2a5a1077a5e049ed2c2c3e2fea82201" -goBinary="/usr/bin/go" -directory=clients -source=cli/main/cli.go -output=clients/linux_amd64/evergreen
[derrida ~/xgen/evergreen](EVG-2159-add-debug-option-to-makefile) % ENABLE_DEBUG=true make -n cli
mkdir -p bin
GOOS="" GOARCH="" go build -o bin/build-cross-compile scripts/build-cross-compile.go
echo go build -o bin/build-cross-compile scripts/build-cross-compile.go
./bin/build-cross-compile -buildName=linux_amd64 -ldflags="-w -s -X=github.com/evergreen-ci/evergreen.BuildRevision=72038235f2a5a1077a5e049ed2c2c3e2fea82201" -goBinary="/usr/bin/go" -directory=clients -source=cli/main/cli.go -output=clients/linux_amd64/evergreen
```

I suppose we should rename the variable for consistency though.

Comment (bsamek) [reply makefile]: -w and -s do the opposite thing that you want though
```
  -s    disable symbol table
  -w    disable DWARF generation
```

**Comment by tychoish** [comment]
fixed.

**Pull request merged**